api/handlers.go

```diff
@@ -296,51 +287,102 @@ func (api *API) userPUT(w http.ResponseWriter, req *http.Request, _ httprouter.P
 	}
 	defer func() { _ = req.Body.Close() }()
 	payload := struct {
-		StripeID string `json:"stripeCustomerId"`
+		Email    string `json:"email,omitempty"`
+		StripeID string `json:"stripeCustomerId,omitempty"`
 	}{}
 	err = json.Unmarshal(bodyBytes, &payload)
 	if err != nil {
 		err = errors.AddContext(err, "failed to parse request body")
 		api.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
-	if payload.StripeID == "" {
-		err = errors.AddContext(err, "empty stripe id")
-		api.WriteError(w, err, http.StatusBadRequest)
+	if payload.Email == "" && payload.StripeID == "" {
+		// The payload is empty, nothing to do.
+		api.WriteSuccess(w)
 		return
 	}
-	// Check if this user already has this ID assigned to them.
-	if payload.StripeID == u.StripeID {
-		// Nothing to do.
-		api.WriteJSON(w, u)
+	// Verify that no other user owns this StripeID.
+	if payload.StripeID != "" {
```

Comment:
Yes, they are.

The `userPUT` handler was initially created to update the Stripe ID of the user. Now that we want to manage their email as well, it can be used for that, too. But none of those is required.

I was checking if the StripeID/Email was empty twice because I was trying to avoid fetching things from the DB too soon but that is making the method confusing. I'll rework it a little bit in order to make it more readable.